specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2020-09-18/kusto.json

```diff
@@ -0,0 +1,4351 @@
+        "type": {
+          "description": "Database principal type.",
+          "type": "string",
+          "enum": [
+            "App",
+            "Group",
+            "User"
+          ],
+          "x-ms-enum": {
+            "name": "DatabasePrincipalType",
+            "modelAsString": true
+          }
+        },
+        "fqn": {
+          "description": "Database principal fully qualified name.",
+          "type": "string"
+        },
+        "email": {
+          "description": "Database principal email if exists.",
+          "type": "string"
+        },
+        "appId": {
+          "description": "Application id - relevant only for application principal type.",
+          "type": "string"
+        },
+        "tenantName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The tenant name of the principal"
+        }
+      },
+      "description": "A class representing database principal entity."
+    },
+    "DatabasePrincipalProperties": {
+      "type": "object",
+      "required": [
+        "principalId",
+        "role",
+        "principalType"
+      ],
+      "properties": {
+        "principalId": {
+          "description": "The principal ID assigned to the database principal. It can be a user email, application ID, or security group name.",
+          "type": "string"
+        },
+        "role": {
+          "description": "Database principal role.",
+          "type": "string",
+          "enum": [
+            "Admin",
+            "Ingestor",
+            "Monitor",
+            "User",
+            "UnrestrictedViewers",
+            "Viewer"
+          ],
+          "x-ms-enum": {
+            "name": "DatabasePrincipalRole",
+            "modelAsString": true
+          }
+        },
+        "tenantId": {
+          "type": "string",
+          "description": "The tenant id of the principal"
+        },
+        "principalType": {
+          "description": "Principal type.",
+          "type": "string",
+          "enum": [
+            "App",
+            "Group",
+            "User"
+          ],
+          "x-ms-enum": {
+            "name": "PrincipalType",
+            "modelAsString": true
+          }
+        },
+        "tenantName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The tenant name of the principal"
+        },
+        "principalName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The principal name"
+        },
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The provisioned state of the resource.",
+          "enum": [
+            "Running",
+            "Creating",
+            "Deleting",
+            "Succeeded",
+            "Failed",
+            "Moving"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          }

```

Comment:
There appears to be several same definition of "provisioningState". It might better to put it to definition and share the reference, like https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2020-06-01/network.json#L241-L255